src/overlays/actors/ovl_En_Sth2/z_en_sth2.c

```diff
@@ -27,22 +30,94 @@ const ActorInit En_Sth2_InitVars = {
+    Actor_SetScale(&this->actor, 0.01f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 36.0f);
+    this->unused = 0;
+
+    if (play->actorCtx.unk5 & 2) {
+        this->actor.flags |= (ACTOR_FLAG_10 | ACTOR_FLAG_20);
+    } else {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+    this->actionFunc = EnSth2_UpdateSkelAnime;
+}
+
+void EnSth2_Destroy(Actor* thisx, PlayState* play) {
+}
+
+void EnSth2_UpdateSkelAnime(EnSth2* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+}
+
+void EnSth2_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnSth2* this = THIS;
+
+    if (Object_IsLoaded(&play->objectCtx, this->objIndex)) {
+        this->actor.objBankIndex = this->objIndex;
+        Actor_SetObjectDependency(play, &this->actor);
+        SkelAnime_InitFlex(play, &this->skelAnime, &object_sth_Skel_0031F8, &gEnSth2WavingHandAnim, this->jointTable,
+                           this->morphTable, OBJECT_STH_LIMB_MAX);
+        Animation_PlayLoop(&this->skelAnime, &gEnSth2WavingHandAnim);
+        this->actor.update = EnSth2_UpdateActionFunc;
+        this->actor.draw = EnSth2_Draw;
+    }
+}
+
+void EnSth2_UpdateActionFunc(Actor* thisx, PlayState* play) {
+    EnSth2* this = THIS;
+
+    this->actionFunc(this, play);
+}
+
+s32 EnSth2_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    s32 pad;
+
+    if (limbIndex == OBJECT_STH_LIMB_0F) {
+        *dList = gEnSth2HeadDL;
+    }
+    if ((limbIndex == OBJECT_STH_LIMB_08) || (limbIndex == OBJECT_STH_LIMB_0A) || (limbIndex == OBJECT_STH_LIMB_0D)) {
+        rot->y += (s16)(Math_SinS((play->state.frames * ((limbIndex * 0x32) + 0x814))) * 200.0f);
+        rot->z += (s16)(Math_CosS((play->state.frames * ((limbIndex * 0x32) + 0x940))) * 200.0f);
+    }
+    return false;
+}
 
-extern UNK_TYPE D_060031F8;
+void EnSth2_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Vec3f D_80BFA010 = { 700.0f, 400.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sth2/EnSth2_Init.s")
+    if (limbIndex == OBJECT_STH_LIMB_0F) {
+        Matrix_MultVec3f(&D_80BFA010, &thisx->focus.pos);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sth2/EnSth2_Destroy.s")
+        OPEN_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sth2/func_80BF75A0.s")
+        gSPDisplayList(POLY_OPA_DISP++, gEnSth2HairDL);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sth2/EnSth2_Update.s")
+        CLOSE_DISPS(play->state.gfxCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Sth2/func_80BF7688.s")
+void EnSth2_Draw(Actor* thisx, PlayState* play2) {
+    static Color_RGB8 sEnvColors[] = { { 190, 110, 0 },  { 0, 180, 110 },   { 0, 255, 80 },
+                                       { 255, 160, 60 }, { 190, 230, 250 }, { 240, 230, 120 } };
```

Comment:
```suggestion
    static Color_RGB8 sEnvColors[] = { { 190, 110, 0 },  { 0, 180, 110 },   { 0, 255, 80 },
                                       { 255, 160, 60 }, { 190, 230, 250 }, { 240, 230, 120 }, };
```
You still want the trailing comma.